system/handlers/core/AssetDownload.cfc

```diff
@@ -21,6 +21,11 @@ component {
 		var config            = assetManagerService.getDerivativeConfig( assetId );
 		var configHash        = assetManagerService.getDerivativeConfigHash( config );
 		var queueEnabled      = isFeatureEnabled( "assetQueue" );
+		var assetPublicUrl    = assetManagerService.getAssetUrl(
```

Comment:
We should only do this if it is not a derivative. Should be moved down to the conditional logic around whether this is a derivative, version or plain asset.

I think we are also missing logic for getting version URL?